src/createServer.js

```diff
@@ -1,11 +1,190 @@
 'use strict';
 
-// const express = require('express');
+const express = require('express');
 
 function createServer() {
-  // Use express to create a server
-  // Add a routes to the server
-  // Return the server (express app)
+  const app = express();
+
+  app.use(express.json());
+
+  const users = [];
+  const expenses = [];
+
+  app.post('/users', (req, res) => {
+    if (!req.body.name) {
+      return res.sendStatus(400);
+    }
+
+    const user = {
+      id: users.length,
+      name: req.body.name,
+    };
+
+    users.push(user);
+
+    res.status(201).send(user);
+  });
+
+  app.get('/users', (req, res) => {
+    res.status(200).send(users);
+  });
+
+  app.get('/users/:id', (req, res) => {
+    const user = users.find(u => u.id === Number(req.params.id));
+
+    if (!user) {
+      return res.sendStatus(404);
+    }
+    res.status(200).send(user);
+  });
+
+  app.delete('/users/:id', (req, res) => {
+    const user = users.find(u => u.id === Number(req.params.id));
+
+    if (!user) {
+      return res.sendStatus(404);
+    }
+
+    const index = users.indexOf(user);
+
+    users.splice(index, 1);
+
+    res.status(204).end();
+  });
+
+  app.patch('/users/:id', (req, res) => {
+    const user = users.find(u => u.id === Number(req.params.id));
+
+    if (!user) {
+      return res.sendStatus(404);
+    }
+
+    if (req.body.name) {
+      user.name = req.body.name;
+    }
+
+    res.status(200).send(user);
+  });
```

Comment:
how about creating a router for those routes? then you could separatem them into 2 files and only import it in create server function 😄